frontend/src/metabase/meta/Parameter.js

```diff
@@ -356,19 +455,71 @@ export function parameterToMBQLFilter(
     const field = metadata.field(fieldId);
     // if the field is numeric, parse the value as a number
     if (isNumericBaseType(field)) {
-      return numberParameterValueToMBQL(parameter.value, fieldRef);
+      return numberParameterValueToMBQL(parameter, fieldRef);
     } else {
-      return stringParameterValueToMBQL(parameter.value, fieldRef);
+      return stringParameterValueToMBQL(parameter, fieldRef);
     }
   }
 }
 
 export function getParameterIconName(parameterType: ?ParameterType) {
-  if (/^date\//.test(parameterType || "")) {
-    return "calendar";
-  } else if (/^location\//.test(parameterType || "")) {
-    return "location";
-  } else {
-    return "label";
+  const [type] = splitType(parameterType);
+  switch (type) {
+    case "date":
+      return "calendar";
+    case "location":
+      return "location";
+    case "category":
+      return "string";
+    case "number":
+      return "number";
+    case "id":
+    default:
+      return "label";
+  }
+}
+
+export function mapUITypeToParameterType(parameter) {
```

Comment:
There's a decent amount of code on the FE that expects `parameter.type` to be unique, so this is my workaround, to replace `location` and `category` with `string` to map to what the backend expects the type to be. It feels a little gross, but here are the benefits of this approach:

1. It's an easy way to handle existing `parameters` with types like `category` and `location/city`. They are both converted to `string/=`.
2. We will need some sort of function like this in the near future to make the `all-options` widget work.